File: dotnet/src/Pactman.NonprofitCheckPlus/SdkVersion.cs
namespace Pactman.NonprofitCheckPlus
{
    /// <summary>
    /// The SDK version, reported in the <c>User-Agent</c> header.
    /// </summary>
    /// <remarks>
    /// Kept in sync with the <c>PactmanSdkVersion</c> property in
    /// <c>Directory.Build.props</c> by a unit test rather than a build step, so the
    /// value is a compile-time constant instead of an assembly-metadata lookup at
    /// runtime.
    /// </remarks>
    public static class SdkVersion
    {
        /// <summary>The release version, reported in the <c>User-Agent</c> header.</summary>
        public const string Version = "1.0.0";

        /// <summary>The NuGet package identifier, reported in the <c>User-Agent</c> header.</summary>
        public const string PackageName = "Pactman.NonprofitCheckPlus";
    }
}

File: java/sdk/src/main/java/org/pactman/nonprofitcheckplus/SdkVersion.java
package org.pactman.nonprofitcheckplus;

import java.util.Locale;

/**
 * The SDK version, reported in the {@code User-Agent} header.
 *
 * <p>Held as a compile-time constant rather than read from the jar manifest at
 * runtime: a manifest lookup returns nothing once the SDK is shaded into a fat
 * jar, which is exactly the deployment where knowing the version matters. A unit
 * test holds this against the POM so the two cannot drift.
 */
public final class SdkVersion {

    /** The released version of this package. */
    public static final String VERSION = "1.0.0";

    /** The distribution name, reported in the {@code User-Agent} header. */
    public static final String PACKAGE_NAME = "pactman-nonprofit-check-plus";

    private SdkVersion() {
    }

    /**
     * The {@code User-Agent} this SDK sends.
     *
     * <p>Formatted as
     * {@code pactman-nonprofit-check-plus/<version> (java/<version>; <os>)}, so
     * a server-side log can tell which SDK and which runtime made a call.
     *
     * @return the user agent string.
     */
    public static String userAgent() {
        String java = System.getProperty("java.version", "unknown");
        String os = System.getProperty("os.name", "unknown");

        return String.format(
                Locale.ROOT, "%s/%s (java/%s; %s)", PACKAGE_NAME, VERSION, java, os);
    }
}
